packages/eslint-plugin/src/rules/unified-signatures.ts

```diff
@@ -591,6 +615,16 @@ function getOverloadInfo(node: OverloadNode): string {
   }
 }
 
+function getStaticParameterName(param: TSESTree.Node): string | undefined {
+  switch (param.type) {
+    case AST_NODE_TYPES.Identifier:
+      return param.name;
+    case AST_NODE_TYPES.RestElement:
+      return getStaticParameterName(param.argument);
+    default:
```

Comment:
quick note that these is valid:
```ts
function foo({a}: {a: string}): void;
function foo([a]: [a: string]): void;
```

I think it's fine for us to ignore these cases as they're rare and pretty dumb.